keps/sig-scheduling/2926-job-mutable-node-affinity/README.md

```diff
@@ -0,0 +1,832 @@
+the same zone, or all either on GPU model x or y, but not a mix of both.
+
+We made the first step towards achieving those semantics by introducing the `suspend`
+flag to the Job API, which allowed a queue controller to decide when a job should start.
+The idea was that such a controller would unsuspend the job when specific constraints
+are met (e.g., capacity is provisioned). However, once a job is unsuspended, the queue 
+controller has no influence on where the pods of a job will actually land.
+
+Adding the ability to mutate a job's node affinity constraints gives a queue controller
+the ability to influence placement while at the same time offloading actual pod-to-node
+assignment to the existing kube-scheduler.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- Allow mutating node affinity constraints of suspended jobs.
+
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- Allow mutating node affinity of running jobs.
+- Allow mutating node affinity of pods.
+- Allow mutating node affinity of apps other than jobs.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+The proposal is to relax validation of suspended jobs' update to the pod template's
+node affinity constraints.
+
+This has no impact on the job-controller. The job controller has no dependency
+on the node affinity constraints expressed in a job's pod template.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+<!--
+
+### User Stories (Optional)
+
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+-->
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
```

Comment:
Shall we mention something like: it's recommended for the job queue controller to update the Job's nodeaffinity along with unsuspending the job, i.e., in one API call?